### ISSUE TITLE
error[E0623]: lifetime mismatch

### ISSUE THREAD

**Issue by Dhamma-valence** [crev-dev/cargo-crev#469]
Thank you for using `crev`!

If you want to ask a question, try crev's gitter channel first: https://gitter.im/dpc/crev . You might get a faster answer.

```
RUST_BACKTRACE=1 cargo install cargo-crev
```
Results in the following output:
```
    Updating crates.io index
  Installing cargo-crev v0.23.0
   Compiling libc v0.2.117
   Compiling proc-macro2 v1.0.36
   Compiling unicode-xid v0.2.2
   Compiling cfg-if v1.0.0
   Compiling syn v1.0.86
   Compiling autocfg v1.1.0
   Compiling version_check v0.9.4
   Compiling memchr v2.4.1
   Compiling pkg-config v0.3.24
   Compiling serde v1.0.136
   Compiling serde_derive v1.0.136
   Compiling log v0.4.14
   Compiling typenum v1.15.0
   Compiling lazy_static v1.4.0
   Compiling crossbeam-utils v0.8.7
   Compiling fnv v1.0.7
   Compiling bitflags v1.3.2
   Compiling tinyvec_macros v0.1.0
   Compiling matches v0.1.9
   Compiling regex-syntax v0.6.25
   Compiling pin-project-lite v0.2.8
   Compiling ryu v1.0.9
   Compiling serde_json v1.0.78
   Compiling getrandom v0.1.16
   Compiling itoa v1.0.1
   Compiling futures-core v0.3.21
   Compiling siphasher v0.3.9
   Compiling hashbrown v0.11.2
   Compiling ppv-lite86 v0.2.16
   Compiling percent-encoding v2.1.0
   Compiling once_cell v1.9.0
   Compiling unicode-bidi v0.3.7
   Compiling openssl-probe v0.1.5
   Compiling subtle v2.4.1
   Compiling futures-channel v0.3.21
   Compiling foreign-types-shared v0.1.1
   Compiling futures-task v0.3.21
   Compiling openssl v0.10.38
   Compiling futures-sink v0.3.21
   Compiling bytes v1.1.0
   Compiling strsim v0.10.0
   Compiling opaque-debug v0.3.0
   Compiling crc32fast v1.3.2
   Compiling ident_case v1.0.1
   Compiling scopeguard v1.1.0
   Compiling futures-util v0.3.21
   Compiling regex-automata v0.1.10
   Compiling pin-utils v0.1.0
   Compiling futures-io v0.3.21
   Compiling slab v0.4.5
   Compiling ucd-trie v0.1.3
   Compiling semver v1.0.5
   Compiling crossbeam-epoch v0.9.7
   Compiling same-file v1.0.6
   Compiling base64 v0.13.0
   Compiling unic-char-range v0.9.0
   Compiling maplit v1.0.2
   Compiling unic-common v0.9.0
   Compiling cpufeatures v0.2.1
   Compiling encoding_rs v0.8.30
   Compiling either v1.6.1
   Compiling hex v0.4.3
   Compiling anyhow v1.0.53
   Compiling arrayvec v0.5.2
   Compiling native-tls v0.2.8
   Compiling shell-escape v0.1.5
   Compiling linked-hash-map v0.5.4
   Compiling httparse v1.5.1
   Compiling rayon-core v1.9.1
   Compiling itoa v0.4.8
   Compiling deunicode v0.4.3
   Compiling try-lock v0.2.3
   Compiling curl v0.4.42
   Compiling unicode-width v0.1.9
   Compiling byteorder v1.4.3
   Compiling signature v1.5.0
   Compiling vec_map v0.8.2
   Compiling parking_lot_core v0.8.5
   Compiling humantime v2.1.0
   Compiling ansi_term v0.12.1
   Compiling strsim v0.8.0
   Compiling arrayvec v0.4.12
   Compiling rprompt v1.0.5
   Compiling httpdate v1.0.2
   Compiling humansize v1.1.1
   Compiling termcolor v1.1.2
   Compiling tower-service v0.3.1
   Compiling fastrand v1.7.0
   Compiling utf8parse v0.2.0
   Compiling remove_dir_all v0.5.3
   Compiling constant_time_eq v0.1.5
   Compiling nodrop v0.1.14
   Compiling unicode-segmentation v1.9.0
   Compiling crossbeam-queue v0.3.4
   Compiling arrayref v0.3.6
   Compiling smallvec v1.8.0
   Compiling hex v0.3.2
   Compiling ipnet v2.3.1
   Compiling bytecount v0.6.2
   Compiling default v0.1.2
   Compiling half v1.8.2
   Compiling mime v0.3.16
   Compiling num_threads v0.1.3
   Compiling home v0.5.3
   Compiling glob v0.3.0
   Compiling fixedbitset v0.4.1
   Compiling resiter v0.4.0
   Compiling lazycell v1.3.0
   Compiling bytesize v1.1.0
   Compiling rustc-workspace-hack v1.0.0
   Compiling instant v0.1.12
   Compiling generic-array v0.14.5
   Compiling uncased v0.9.6
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling im-rc v15.0.0
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling indexmap v1.8.0
   Compiling memoffset v0.6.5
   Compiling rayon v1.5.1
   Compiling tracing-core v0.1.22
   Compiling tinyvec v1.5.1
   Compiling form_urlencoded v1.0.1
   Compiling thread_local v1.1.4
   Compiling foreign-types v0.3.2
   Compiling lock_api v0.4.6
   Compiling http v0.2.6
   Compiling pest v2.1.3
   Compiling walkdir v2.3.2
   Compiling unic-char-property v0.9.0
   Compiling unic-ucd-version v0.9.0
   Compiling itertools v0.10.3
   Compiling yaml-rust v0.4.5
   Compiling slug v0.1.4
   Compiling textwrap v0.11.0
   Compiling ed25519 v1.3.0
   Compiling blake2b_simd v0.5.11
   Compiling heck v0.3.3
   Compiling tracing v0.1.30
   Compiling unic-ucd-segment v0.9.0
   Compiling unicode-normalization v0.1.19
   Compiling pest_meta v2.1.3
   Compiling http-body v0.4.4
   Compiling aho-corasick v0.7.18
   Compiling bstr v0.2.17
   Compiling grep-matcher v0.1.5
   Compiling num_cpus v1.13.1
   Compiling socket2 v0.4.4
   Compiling getrandom v0.2.4
   Compiling atty v0.2.14
   Compiling dirs-sys-next v0.1.2
   Compiling filetime v0.2.15
   Compiling memmap2 v0.3.1
   Compiling jobserver v0.1.24
   Compiling tempfile v3.3.0
   Compiling dirs-sys v0.3.6
   Compiling term_size v0.3.2
   Compiling time v0.3.7
   Compiling rpassword v5.0.1
   Compiling quote v1.0.15
   Compiling mio v0.7.14
   Compiling want v0.3.0
   Compiling crossbeam-channel v0.5.2
   Compiling rust-argon2 v0.8.3
   Compiling bitmaps v2.1.0
   Compiling unic-segment v0.9.0
   Compiling idna v0.2.3
   Compiling num-format v0.4.0
   Compiling phf_shared v0.10.0
   Compiling encoding_rs_io v0.1.7
   Compiling opener v0.5.0
   Compiling regex v1.5.4
   Compiling rand_core v0.5.1
   Compiling clap v2.34.0
   Compiling rand_core v0.6.3
   Compiling directories-next v2.0.0
   Compiling dirs-next v2.0.0
   Compiling tar v0.4.38
   Compiling parking_lot v0.11.2
   Compiling dirs v3.0.2
   Compiling cc v1.0.72
   Compiling vte_generate_state_changes v0.1.1
   Compiling petgraph v0.6.0
   Compiling tokio v1.16.1
   Compiling digest v0.9.0
   Compiling cipher v0.3.0
   Compiling crypto-mac v0.8.0
   Compiling block-buffer v0.9.0
   Compiling dbl v0.3.2
   Compiling aead v0.4.3
   Compiling phf v0.10.1
   Compiling grep-searcher v0.1.8
   Compiling globset v0.4.8
   Compiling env_logger v0.8.4
   Compiling env_logger v0.9.0
   Compiling sized-chunks v0.6.5
   Compiling rand_chacha v0.2.2
   Compiling rand_xoshiro v0.4.0
   Compiling rand_chacha v0.3.1
   Compiling parse-zoneinfo v0.3.0
   Compiling term v0.7.0
   Compiling openssl-src v111.17.0+1.1.1m
   Compiling libz-sys v1.1.3
   Compiling libnghttp2-sys v0.1.7+1.45.0
   Compiling libssh2-sys v0.2.23
   Compiling curl-sys v0.4.52+curl-7.81.0
   Compiling libgit2-sys v0.12.26+1.3.0
   Compiling vte v0.10.1
   Compiling crossbeam-deque v0.8.1
   Compiling crypto-mac v0.11.1
   Compiling aes v0.7.5
   Compiling ctr v0.8.0
   Compiling blake2 v0.9.2
   Compiling sha2 v0.9.9
   Compiling ignore v0.4.18
   Compiling syn-inline-mod v0.5.0
   Compiling synstructure v0.12.6
   Compiling darling_core v0.12.4
   Compiling pest_generator v2.1.3
   Compiling tokio-util v0.6.9
   Compiling rand v0.7.3
   Compiling rand v0.8.4
   Compiling futures-macro v0.3.21
   Compiling thiserror-impl v1.0.30
   Compiling typed-builder v0.9.1
   Compiling openssl-sys v0.9.72
   Compiling cmac v0.6.0
   Compiling crossbeam v0.8.1
   Compiling strip-ansi-escapes v0.1.1
   Compiling zeroize_derive v1.3.1
   Compiling structopt-derive v0.4.18
   Compiling pest_derive v2.1.0
   Compiling darling_macro v0.12.4
   Compiling globwalk v0.8.1
   Compiling thiserror v1.0.30
   Compiling phf_generator v0.10.0
   Compiling zeroize v1.3.0
   Compiling darling v0.12.4
   Compiling flate2 v1.0.22
   Compiling crev-recursive-digest v0.5.0
   Compiling phf_codegen v0.10.0
   Compiling curve25519-dalek v3.2.0
   Compiling aes-siv v0.6.2
   Compiling derive_builder_core v0.10.2
   Compiling structopt v0.3.26
   Compiling chrono-tz-build v0.0.2
   Compiling cargo v0.59.0
   Compiling h2 v0.3.11
   Compiling futures-executor v0.3.21
   Compiling derive_builder_macro v0.10.2
   Compiling chrono-tz v0.6.1
   Compiling futures v0.3.21
   Compiling derive_builder v0.10.2
   Compiling url v2.2.2
   Compiling chrono v0.4.19
   Compiling serde_yaml v0.8.23
   Compiling ed25519-dalek v1.0.1
   Compiling toml v0.5.8
   Compiling serde_urlencoded v0.7.1
   Compiling cargo-platform v0.1.2
   Compiling serde_cbor v0.11.2
   Compiling os_info v3.2.0
   Compiling serde_ignored v0.1.2
   Compiling dashmap v4.0.2
   Compiling cargo-geiger-serde v0.2.1
   Compiling rustfix v0.6.0
   Compiling crev-common v0.23.0
   Compiling hyper v0.14.16
   Compiling geiger v0.4.8
   Compiling crev-data v0.23.0
   Compiling crev-wot v0.23.0
error[E0623]: lifetime mismatch
   --> /home/valence/.cargo/registry/src/github.com-1ecc6299db9ec823/crev-wot-0.23.0/src/lib.rs:980:10
    |
978 |         &'s self,
    |         --------
979 |         id: &'id Id,
    |             ------- these two types are declared with different lifetimes...
980 |     ) -> impl Iterator<Item = (&'id Id, TrustLevel)> + 's {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...but data from `id` flows into `self` here

For more information about this error, try `rustc --explain E0623`.
error: could not compile `crev-wot` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-crev v0.23.0`, intermediate artifacts can be found at `/tmp/cargo-installLFFDJc`

Caused by:
  build failed
```
cargo-crev 0.22.2

I installed crev via ```cargo install cargo-crev```

I am using Debian GNU/Linux bookworm/sid, and my current kernel is 5.21-xanmod1-tt

**Comment by dpc** [comment]
`rustc --version`? Probably too old Rust compiler.

**Comment by Dhamma-valence** [comment]
rustc 1.56.0

**Comment by dpc** [comment]
```
    Finished release [optimized] target(s) in 4m 17s
   Replacing /home/dpc/.cargo/bin/cargo-crev
    Replaced package `cargo-crev v0.23.0` with `cargo-crev v0.23.0` (executable `cargo-crev`)
> rustc --version
rustc 1.58.1 (db9d1b20b 2022-01-20)
```

**Comment by dpc** [comment]
```
> rustup run 1.56.0 cargo install --force cargo-crev
...
   Compiling crev-wot v0.23.0
error[E0623]: lifetime mismatch
   --> /home/dpc/.cargo/registry/src/github.com-1ecc6299db9ec823/crev-wot-0.23.0/src/lib.rs:980:10
    |
978 |         &'s self,
    |         --------
979 |         id: &'id Id,
    |             ------- these two types are declared with different lifetimes...
980 |     ) -> impl Iterator<Item = (&'id Id, TrustLevel)> + 's {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...but data from `id` flows into `self` here

For more information about this error, try `rustc --explain E0623`.
error: could not compile `crev-wot` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-crev v0.23.0`, intermediate artifacts can be found at `/tmp/cargo-install5Nn9GV`

Caused by:
  build failed
```

Not sure why, but it looks like we require newer compiler. :shrug:

**Comment by dpc** [comment]
Only recently we fixed the CI and it tests 1.58.0+ https://github.com/crev-dev/cargo-crev/blob/88b96096b7b996ad64f42f05031c871c67bbd21f/.github/workflows/ci.yml#L48
